apps/smart-contracts/core/test/DepositHook.test.ts

```diff
@@ -154,18 +165,63 @@ describe('=> DepositHook', () => {
       expect(await depositHook.depositsAllowed()).to.eq(false)
 
       await expect(
-        depositHook.connect(vault).hook(user.address, TEST_AMOUNT_BEFORE_FEE, TEST_AMOUNT_AFTER_FEE)
+        depositHook
+          .connect(collateralSigner)
+          .hook(user.address, TEST_AMOUNT_BEFORE_FEE, TEST_AMOUNT_AFTER_FEE)
       ).to.revertedWith('deposits not allowed')
     })
 
-    it('should call recordDeposit with the correct parameters', async () => {
-      expect(TEST_AMOUNT_BEFORE_FEE).to.not.eq(TEST_AMOUNT_AFTER_FEE)
+    it('calls recordDeposit if no fee', async () => {
+      await depositHook
+        .connect(collateralSigner)
+        .hook(user.address, TEST_AMOUNT_BEFORE_FEE, TEST_AMOUNT_BEFORE_FEE)
+
+      expect(mockDepositRecord.recordDeposit).to.be.calledWith(user.address, TEST_AMOUNT_BEFORE_FEE)
+    })
+
+    it('calls recordDeposit if fee > 0', async () => {
       await depositHook
-        .connect(vault)
+        .connect(collateralSigner)
         .hook(user.address, TEST_AMOUNT_BEFORE_FEE, TEST_AMOUNT_AFTER_FEE)
 
       expect(mockDepositRecord.recordDeposit).to.be.calledWith(user.address, TEST_AMOUNT_AFTER_FEE)
     })
+
+    it('transfers fee to treasury if fee > 0', async () => {
+      expect(TEST_AMOUNT_BEFORE_FEE).to.not.eq(TEST_AMOUNT_AFTER_FEE)
```

Comment:
nit: separate the 'Arrange' section from the 'Act' section. same for other tests
```suggestion
      expect(TEST_AMOUNT_BEFORE_FEE).to.not.eq(TEST_AMOUNT_AFTER_FEE)
      
```